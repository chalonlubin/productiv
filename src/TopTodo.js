import React from "react";
import Todo from "./Todo";

/** Shows the top todo.
 *
 * Props:
 * - todos
 *
 * TodoApp -> TopTodo
 */



// Fix was returning null vs undefined.
function TopTodo({todos}) {
  if (todos.length === 0) return null;

  let top = todos.reduce(
    (acc, cur) => cur.priority < acc.priority ? cur : acc, todos[0]);
    console.log("top:",top)

    return <Todo todo={top} />; // {top: {todo: }} // {key: value}

  // if (todos.length > 0) {
  //   let top = todos.reduce(
  //     // lowest-priority # is the highest priority
  //       (acc, cur) => cur.priority < acc.priority ? cur : acc, todos[0]);
  //       return <Todo  id={top.id} title={top.title} description={top.description} priority={top.priority} />;
  // }
}


export default TopTodo;