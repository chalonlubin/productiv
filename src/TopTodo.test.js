import React from "react";
import { render} from "@testing-library/react";
import TopTodo from "./TopTodo";

describe("TopTodo Tests", function () {

  let todos = [
  {
    id: 1,
    title: "Top-Title",
    description: "Top-Description",
    priority: 1 },
    {
      id: 2,
      title: "Test-Title",
      description: "Test-Description",
      priority: 2 }
  ]

  it("renders without crashing", function () {
    render(<TopTodo todos={todos} />);
  });

  it("contains expected title", function () {
    const result = render(<TopTodo todos={todos} />);
    expect(result.queryByText("Top-Title")).toBeInTheDocument();
  });

  it("matches snapshot", function () {
    const { container } = render(<TopTodo todos={todos}/>);
    expect(container).toMatchSnapshot();
  });

  it("can correctly compare int priority values", function () {
    const result = render(<TopTodo todos={todos} />);
    expect(result.queryByText("Top-Description")).toBeInTheDocument();
    expect(result.queryByText("Test-Description")).not.toBeInTheDocument();
  });

  it("returns null when todos list is empty", function () {
    todos = [];
    const result = render(<TopTodo todos={todos} />);
    expect(result.queryByText("priority")).not.toBeInTheDocument();
  });


});
