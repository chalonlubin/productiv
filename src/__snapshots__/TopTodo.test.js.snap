// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopTodo Tests matches snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        Top-Title
      </b>
       
      <small>
        (priority: 
        1
        )
      </small>
    </div>
    <div>
      <small>
        Top-Description
      </small>
    </div>
  </div>
</div>
`;
